wapitiCore/net/crawler.py

```diff
@@ -926,6 +926,52 @@ def iter_forms(self, autofill=True):
                 )
                 yield new_form
 
+    def find_login_form(self):
+        """Returns the login Request extracted from the Page, the username and password fields."""
+
+        for form in self.soup.find_all("form"):
+            username_field_idx = []
+            password_field_idx = []
+
+            for i, input_field in enumerate(form.find_all("input")):
```

Comment:
![](https://thebeerthrillershome.files.wordpress.com/2020/02/thisistheway.gif)

But you should either make sure input type is text for username field or discard other common types.

For example I tried on a phpBB forum where two fields were incorrectly threated as login fields : 

```
input type=checkbox name=autologin
input type=submit name=login
```